PowerEditor/src/WinControls/FunctionList/functionParser.cpp

```diff
@@ -598,14 +598,14 @@ void FunctionParser::getInvertZones(vector< pair<int, int> > &  destZones, vecto
 {
 	if (sourceZones.size() == 0)
 	{
-		destZones.push_back(pair<int, int>(begin, end));

```

Comment:
> *more on this in a second*

`FunctionParser::getInvertZones` is called from three places:
- [`FunctionMixParser::parse`](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/0774106a40cf55acb6653adefae7fcdde6e581a2/PowerEditor/src/WinControls/FunctionList/functionParser.cpp#L670), declared as: `void FunctionMixParser::parse(std::vector<foundInfo> & foundInfos, size_t begin, size_t end, ScintillaEditView **ppEditView, generic_string classStructName)`
- [`FunctionZoneParser::parse`](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/0774106a40cf55acb6653adefae7fcdde6e581a2/PowerEditor/src/WinControls/FunctionList/functionParser.cpp#L629), declared as: `void FunctionZoneParser::parse(std::vector<foundInfo> & foundInfos, size_t begin, size_t end, ScintillaEditView **ppEditView, generic_string classStructName)`
- [`FunctionUnitParser::parse`](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/0774106a40cf55acb6653adefae7fcdde6e581a2/PowerEditor/src/WinControls/FunctionList/functionParser.cpp#L640), declared as: `void FunctionUnitParser::parse(std::vector<foundInfo> & foundInfos, size_t begin, size_t end, ScintillaEditView **ppEditView, generic_string classStructName)`


All three of which are [virtually called from `FunctionParsersManager::parse`](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/0774106a40cf55acb6653adefae7fcdde6e581a2/PowerEditor/src/WinControls/FunctionList/functionParser.cpp#L443), where `docLen`(`end`) is an `int`, that's apparently coerced to an `int`.

If we change the declaration/definition of `FunctionMixParser::parse`, `FunctionZoneParser::parse`, `FunctionUnitParser::parse`, to accept `begin` & `end` as `int`s, then we can also change `FunctionParser::getInvertZones` to accept `begin` & `end` as `int`s.

As best as I can tell, many internal function calls (in `FunctionMixParser::parse`, `FunctionZoneParser::parse`, and `FunctionUnitParser::parse`) accept `begin` & `end` as `int`s *as it is now*, so this won't require any rippling changes. **But**, we'd avoid all the casting.